anchor/programs/bless-token/src/context/fund.rs

```diff
@@ -0,0 +1,186 @@
+
+    #[account(
+        mut,
+        seeds = [SEED_BLESS_VAULT.as_bytes(), bless_mint.key().as_ref()],
+        bump = bless_state.vault_bump,
+    )]
+    pub vault_account: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == bless_state.mint_pubkey,
+    )]
+    pub bless_mint: Box<Account<'info, Mint>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == preseed_sale.mint,
+        constraint = preseed_sale.key() == bless_state.preseed_sale,
+    )]
+    pub preseed_sale: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == seed_sale.mint,
+        constraint = seed_sale.key() == bless_state.seed_sale,
+    )]
+    pub seed_sale: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == advisors.mint,
+        constraint = advisors.key() == bless_state.advisors,
+    )]
+    pub advisors: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == team.mint,
+        constraint = team.key() == bless_state.team,
+    )]
+    pub team: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == ecosystem.mint,
+        constraint = ecosystem.key() == bless_state.ecosystem,
+    )]
+    pub ecosystem: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == foundation.mint,
+        constraint = foundation.key() == bless_state.foundation,
+    )]
+    pub foundation: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == market_making.mint,
+        constraint = market_making.key() == bless_state.market_making,
+    )]
+    pub market_making: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == tgt_marketing.mint,
+        constraint = tgt_marketing.key() == bless_state.tgt_marketing,
+    )]
+    pub tgt_marketing: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == community_airdrop.mint,
+        constraint = community_airdrop.key() == bless_state.community_airdrop,
+    )]
+    pub community_airdrop: Box<Account<'info, TokenAccount>>,
+
+    #[account(
+        mut,
+        constraint = bless_mint.key() == community_incentives.mint,
+        constraint = community_incentives.key() == bless_state.community_incentives,
+    )]
+    pub community_incentives: Box<Account<'info, TokenAccount>>,
+
+    pub system_program: Program<'info, System>,
+
+    pub token_program: Program<'info, Token>,
+}
+
+impl<'info> FundBlessToken<'info> {
+    pub fn transfer(&self, to: AccountInfo<'info>, amount: u64) -> Result<()> {
+        let cpi_ctx = CpiContext::new(
+            self.bless_mint.to_account_info(),
```

Comment:
Shouldn't this be the token program - which performs the transfer?
```suggestion
            self.token_program.to_account_info(),
```
If so; please do add a test for this; unit tests should catch/test such cases

Reply:
No, the token_program is the system token program needed by contact, bless_mint is the specify token.